doubleclutch30/ArchitectureStudy/app/src/main/java/com/example/architecturestudy/data/source/remote/NaverSearchRemoteDataSourceImpl.kt

```diff
@@ -0,0 +1,90 @@
+import com.example.architecturestudy.network.response.BlogData
+import com.example.architecturestudy.network.response.ImageData
+import com.example.architecturestudy.network.response.KinData
+import com.example.architecturestudy.network.response.MovieData
+import com.example.architecturestudy.network.Api
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class NaverSearchRemoteDataSourceImpl(private val restClient: Api) : NaverSearchRemoteDataSource {
+
+    override fun getMovie(keyword: String, success: (List<MovieItems>) -> Unit, fail: (Throwable) -> Unit) {
+        restClient.requestMovies(keyword).enqueue(object : Callback<MovieData> {
+            override fun onFailure(call: Call<MovieData>, t: Throwable) {
+                fail(t)
+            }
+
+            override fun onResponse(call: Call<MovieData>, response: Response<MovieData>) {
+                if(response.isSuccessful) {
+                    response.body()?.items?.let {
+                        success(
+                            it
+                        )
+                    }
+                }
+            }
+        })
+    }
+
+    override fun getBlog(keyword: String, success: (List<BlogItems>) -> Unit, fail: (Throwable) -> Unit) {
+        restClient.requestBlog(keyword).enqueue(object : Callback<BlogData> {
+            override fun onFailure(call: Call<BlogData>, t: Throwable) {
+                fail(t)
+            }
+
+            override fun onResponse(call: Call<BlogData>, response: Response<BlogData>) {
+                if(response.isSuccessful) {
+                    response.body()?.items?.let {
+                        success(
+                            it
+                        )
+                    }
+                }
+            }
+        })
+    }
+
+    override fun getKin(keyword: String, success: (List<KinItems>) -> Unit, fail: (Throwable) -> Unit) {
+        restClient.requestKin(keyword).enqueue(object : Callback<KinData> {
+            override fun onFailure(call: Call<KinData>, t: Throwable) {
+                fail(t)
+            }
+
+            override fun onResponse(call: Call<KinData>, response: Response<KinData>) {
+                if(response.isSuccessful) {
+                    response.body()?.items?.let {
+                        success(
+                            it
+                        )
+                    }
+                }
+            }
+        })
+    }
+
+    override fun getImage(keyword: String, success: (List<ImageItems>) -> Unit, fail: (Throwable) -> Unit)
+    {
+        restClient.requestImage(keyword).enqueue(object : Callback<ImageData> {
+            override fun onFailure(call: Call<ImageData>, t: Throwable) {
+                fail(t)
+            }
+
+            override fun onResponse(call: Call<ImageData>, response: Response<ImageData>) {
+                if(response.isSuccessful) {
+                    response.body()?.items?.let {
```

Comment:
다른 키워드로 재검색했을 때 리스트가 비어있어도 기존 뷰가 갱신되지 않아요
items 가 null 일 때도 고려해주시면 좋을 것 같아요
```suggestion
                    success(response.body()?.items ?: emptyList())
```